packages/flutter_tools/test/commands.shard/hermetic/build_web_test.dart

```diff
@@ -88,7 +89,9 @@ void main() {
   }));
 
   test('Builds a web bundle - end to end', () => testbed.run(() async {
-    final CommandRunner<void> runner = createTestCommandRunner(BuildCommand());
+    final BuildCommand buildCommand = BuildCommand();
+    applyMocksToCommand(buildCommand);
```

Comment:
Is there any situation where we wouldn't want to do this? Is it possible to add an assert to `createTestCommandRunner()` to check that the mocks have been applied? If there are any, in the cases where mocks shouldn't be supplied, maybe we should have a call separate from `createTestCommandRunner()` that doesn't have the asserts.

Reply:
Basically, "How do we make this less error prone?"

Reply:
I think the best way would be to create a testbed specific command runner and enforce using that via import checks. For integration tests, I imagine we still want the capability to invoke commands directly